New Quirk: Mood swings!

SgtHunk opened the pull request:
_Please give me feedback if you believe something needs to be changed regarding the quirk_

## About The Pull Request

This adds a new Neutral Quirk, called "Mood Swings". It basically serves as a mix between the two mood quirks (with a slightly increased chance to trigger). Both the positive and negative effects have the same chance to trigger.

## Why It's Good For The Game

A neutral quirk that doesn't drastically affect gameplay, but will allow someone to roleplay as someone that has several changes in mood. For when you really don't know what to settle with.

## Changelog
:cl:
add: Added the Neutral Quirk: Mood Swings.
/:cl:

SgtHunk committed:
Adds the thingy

Adds the Mood Swings quirk, blacklists it from being usable with other mood-related quirks

SgtHunk committed:
almost forgot

defines trait in global vars because im dumb

EdgeLordExe commented:
Call it bipolar disorder

EdgeLordExe commented on code/datums/traits/neutral.dm:
```diff
@@ -268,3 +268,18 @@
+	mood_quirk = TRUE
+
+/datum/quirk/bipolar/on_process(delta_time)
+	if(DT_PROB(0.1, delta_time))
+		SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, "depression_mild", /datum/mood_event/depression_mild)
+	if(DT_PROB(0.1, delta_time))
```
Don't you think this might cause both moodlets to happen at once?

SgtHunk replied:
(From what I understand) they're both 2 different probabilities, so the only way it would cause both to happen is if it hit that 0.1 chance twice at the same time
And when I tested it on local (both with a 5% chance and 10% chance for debugging purposes) they didn't trigger at the same time

EdgeLordExe commented on code/datums/traits/neutral.dm:
```diff
@@ -268,3 +268,18 @@
+	if(DT_PROB(0.1, delta_time))
+		SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, "depression_mild", /datum/mood_event/depression_mild)
+	if(DT_PROB(0.1, delta_time))
+		SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, "jolly", /datum/mood_event/jolly)
```
I suggest using custom moodlets for this

SgtHunk replied:
Done

SgtHunk commented:
I settled with "Mood Swings" since it allows for more variety in what someone can pick (Bipolarity is not the only illness that can cause changes in mood), but if a majority agrees I can change the name to "Bipolar Disorder".

wesoda25 commented:
Whatever happened to the mental illness extension pack someone PRed?

EdgeLordExe commented:
> Whatever happened to the mental illness extension pack someone PRed?

that was  mine and im slowly reviwing the edge psychology

wesoda25 commented:
Ah I see. How will mental illness quirks play into that? Will you just integrate them into the system, or keep them separate but related?

SgtHunk committed:
adds (slightly) custom moodlets

Gives the bipolar quirk a different kind of moodlet flavor text, instead of copying the ones from Jolly/Depression

ATH1909 commented on code/datums/mood_events/generic_positive_events.dm:
```diff
@@ -203,3 +203,8 @@
+
+/datum/mood_event/bipolar_happy
+	description = "<span class='nicegreen'>I'm feeling a bit more happy than usual.</span>\n"
+	mood_change = 6
```
is bipolar_happy intended to have half of the magnitude of bipolar_sad?

SgtHunk replied:
Tbh most of it is just the same as jolly/depression (unless I managed to copy it wrong)
I could lower the effect of bipolar_sad, but checking the code, depression mild has a bigger effect
Although lower number might make it easier to pass it off as a neutral quirk

tiramisuapimancer commented:
To be honest, I'm not sure why it was made that you can't take Jolly and Depression at the same time to amount to this. You used to be able to- I have a rare character slot on another server with Jolly/Depression that was made before they ported mutual quirk blacklists (or at least the jolly/depression one).

Rohesie commented:
> 
> 
> I settled with "Mood Swings" since it allows for more variety in what someone can pick (Bipolarity is not the only illness that can cause changes in mood), but if a majority agrees I can change the name to "Bipolar Disorder".

Mood swings is better. There's too much other stuff associated with the actual disorder, and I don't think it'd be a good idea to add that.

That said, I don't like this feature. I don't see how does this aid RP, nor why it'd be positive to make mechanics for moody characters that swing between euphoria and depression.

Rohesie commented:
We have a lot PRs up and I don't see what this brings on the table. I'd rather not add extra incentives for non-antag unhinged behavior, nor see the positive usage of this in other contexts.
Try getting interest from another maintainer if you want to re-open this PR.
